### ISSUE TITLE
Implement Inline Methods

### ISSUE THREAD

**Issue by mmstick** [redox-os/ion#389]
I've decided that we will pursue implementing this on top of our existing method syntax. The rules are simple, but might require some complex brain work to get the parser ready and capable of parsing the additional rule. The purpose of inline methods is to prevent the need to round trip results to variables and increasing the number of lines of code to parse and execute, costing time and energy, of which we have limited amounts of both.

Basically, it works as follows: if the input parameter in a method begins with `$`, `@`, `[`, `"`, or `'`, then that parameter is an expression and should be parsed as such. Otherwise, treat the parameter as the preferred variable type.

# As Expression

```ion
let lines = @lines($read("filename"))
let lines = @lines($(cmd args...))
let lines = @lines($string_var)
```

# As Variable

```ion
let lines = @lines(string_var)
```

**Comment by huntergoldstein** [comment]
Do we want this as part of https://github.com/redox-os/ion/milestone/1?

**Comment by mmstick** [comment]
I don't think it's a requirement, but it would be a nice bonus point.

**Comment by huntergoldstein** [comment]
Working on implementing this now; is there a reason that there's a separate `if contains_brace { ...` block?

**Comment by mmstick** [comment]
Thanks for attempting this! Words are collected beforehand, so if a braced word is found, that check will handle the creation of the permutation inputs from the given words. It's because an input could be like `ABC{123,456}DEF{123,{abc,def}}ABC`. You'd get a collection of Normal + Brace words, and you need to know all of these connected words in advance before you start permutation them.
